### PR TITLE
Add update board functionality

### PR DESCRIPTION
**Pull request by jessat**
This commit adds the update board functionality.
The tests are also updated accordingly.

Signed-off-by: jessat 14183940+jessat@users.noreply.github.com

### CODE REVIEW

**Comment by jessat** [comment]
@vchrombie what do you think I should do here?  I unsuccessfully tried:
git commit --amend --author="jessat <[14183940+jessat@users.noreply.github.com>"
git push origin add-update-board -f

**Comment by vchrombie** [comment]
Hi @jessat, thanks for the PR.

The problem is with the sign off. It has to be in the format of
```
Signed-off-by: Name <Email>
```

The email has to be enclosed within the `<` `>`. Your sign off of the two commits looks like this.
```
Signed-off-by: jessat 14183940+jessat@users.noreply.github.com
```

You can fix it by
```
git checkout add-update-board
git reset --soft HEAD~2
git commit -s -m "feat: Add update board functionality"
git fetch origin
git rebase master
git push origin add-update-board -f
```

Let me know if you need any help.

File: tests/test_board.py [hunk 1]
```diff
@@ -74,3 +74,15 @@ def test_delete_board(mock_http_client):
...
     assert response.text == '{"_value":null}'
+
+
+# Test for update_board method
+def test_update_board(mock_http_client):
+    board_data = load_mock_data(BOARD_MOCK_DATA)
+    mock_http_client.put.return_value = board_data
+
+    board_manager = BoardManager(mock_http_client)
+    board = board_manager.update_board("test_board_id", "name", "desc")
```
Comment (vchrombie) [inline tests/test_board.py:85]:
```suggestion
    board = board_manager.update_board("test_board_id")
```

File: tests/test_board.py [hunk 2]
```diff
@@ -74,3 +74,15 @@ def test_delete_board(mock_http_client):
...
+
+    assert isinstance(board, Board)
+    assert board.id == board_data["id"]
```
Comment (vchrombie) [inline tests/test_board.py:88]:
```suggestion
    assert board.id == board_data["id"]
    mock_http_client.put.assert_called_once_with("boards/test_board_id/", {})
```

File: pytrello2/board.py [hunk 3]
```diff
@@ -48,3 +48,13 @@ def delete_board(self, board_id):
...
+
+    def update_board(self, board_id, name, desc):
+        """
+        Updates a Board object
+        """
+        data = {}
+        data["name"] = name
+        data["desc"] = desc
+        board_data = self.client.put("boards/", data)
+        return Board(board_data)
```
Comment (vchrombie) [inline pytrello2/board.py:60]: We started using `**kwargs` for the non-required parameters.
You can check the recent [implementation in `card.py`](https://github.com/pytrello2/pytrello2/blob/34ad7738e4321953f5df494afaa92d22fc485f98/pytrello2/card.py#L39).

File: pytrello2/board.py [hunk 4]
```diff
@@ -48,3 +48,13 @@ def delete_board(self, board_id):
         Deletes a Board object with the given ID.
         """
         return self.client.delete(f"boards/{board_id}")
+
+    def update_board(self, board_id, name, desc):
+        """
+        Updates a Board object
+        """
```
Comment (vchrombie) [inline pytrello2/board.py:55]: You can maybe add the parameters in the doc string to follow the coding style.

**Review by vchrombie — CHANGES_REQUESTED** [review]
Just a few more changes before we get in the PR.

You can also rebase to fetch the latest changes.

**Comment by jessat** [comment]
I created a new PR so it is cleaner.

**Pull request closed without merging**